src/applications/static-pages/subscribeComponentAnalyticsEvents.js

```diff
@@ -0,0 +1,50 @@
+import recordEvent from 'platform/monitoring/record-event';
+
+const analyticsEvents = [
+  {
+    componentName: 'Modal',
+    prefix: 'modal',
+    actions: [{ action: 'show', name: 'int-modal-click' }],
+  },
+  {
+    componentName: 'AdditionalInfo',
+    prefix: 'additional-info',
+    actions: [
+      { action: 'expand', name: 'int-additional-info-expand' },
+      { action: 'collapse', name: 'int-additional-info-collapse' },
+    ],
+  },
+];
+
+export default function subscribeComponentAnalyticsEvents() {
+  document.body.addEventListener('component-library-analytics', e => {
+    // Is it a component we are tracking?
+    const component = analyticsEvents.find(
+      c => c.componentName === e.detail.componentName,
+    );
+
+    // Is it an action we are tracking?
+    if (component && component.actions) {
+      const action = component.actions.find(
+        ev => ev.action === e.detail.action,
+      );
+
+      if (action) {
+        const dataLayer = { event: action.name };
+
+        // If the event included additional details / context...
+        if (e.detail.details) {
+          for (const key of Object.keys(e.detail.details)) {
+            const newKey = component.prefix
+              ? `${component.prefix}-${key}`
+              : key;
```

Comment:
If `prefix` is just going to be the `componentName` but in kebab case, why do we need to define it as an additional property?

Reply:
I guess in case - for GTM reasons - it needs to be different.  Such as pointing two different components - ProgressBar and SegmentedProgressBar to the same `progress-bar` prefix.

Should I run it through lodash's `kebabCase` by default and only add the `prefix` param if I need to override?

Reply:
In the `ProgressBar` v. `SegmentedProgressBar` example, what events might they be sending that we would want to capture? I imagine that any updates to a progress bar would first be triggered by a button click which could be responsible for sending an appropriate `nextPage` event or something along those lines.

Yeah, unless there's a strong reason for having a `prefix` I would rely on `kebabCase`.